### DISCUSSION TITLE
Stream load has a Connection reset exception after enabling http v2. Solution

### DISCUSSION THREAD

**Discussion by hf200012** [apache/doris#6732] [RFC]
Stream load has a Connection reset exception after enabling http v2. Solution

In the community version 0.14.0 and before and the Baidu release version before 0.14.13, the connection reset exception occurs after Http V2 is enabled, because the built-in web container is tomcat, and Tomcat is pitted in 307 (Temporary Redirect) , There is a problem with the implementation of this protocol. All in the case of using Stream load to import large amounts of data, the connect reset exception will occur. This is because tomcat started data transmission before the 307 jump, which caused the BE to receive When the data request arrived, the authentication information was missing. Later, the built-in container was changed to Jetty to solve this problem. If you encounter this problem, please upgrade your Doris or disable Http V2 (`enable_http_server_v2=false`).

After the upgrade, also upgrade the http client version of your program to `4.5.13`, and introduce the following dependencies in your pom.xml file

```xml
    <dependency>
      <groupId>org.apache.httpcomponents</groupId>
      <artifactId>httpclient</artifactId>
      <version>4.5.13</version>
    </dependency>
```

在社区版 0.14.0 及之前的版本和百度发行版0.14.13之前的版本，在启用Http V2之后出现connection reset异常，因为Web 容器内置的是tomcat，Tomcat 在 307 (Temporary Redirect) 是有坑的，对这个协议实现是有问题的，所有在使用Stream load 导入大数据量的情况下会出现connect  reset异常，这个是因为tomcat在做307跳转之前就开始了数据传输，这样就造成了BE收到的数据请求的时候缺少了认证信息，之后将内置容器改成了Jetty解决了这个问题，如果你遇到这个问题，请升级你的Doris或者禁用Http V2（`enable_http_server_v2=false`）。

升级以后同时升级你程序的http client 版本到 `4.5.13`，在你的pom.xml文件中引入下面的依赖

```xml
    <dependency>
      <groupId>org.apache.httpcomponents</groupId>
      <artifactId>httpclient</artifactId>
      <version>4.5.13</version>
    </dependency>
```

**Comment by morningman** [comment]
http://doris.incubator.apache.org/master/zh-CN/administrator-guide/load-data/stream-load-manual.html#%E5%B8%B8%E8%A7%81%E9%97%AE%E9%A2%98
